python/BioSimSpace/FreeEnergy/_relative.py

```diff
@@ -417,7 +675,292 @@ def _analyse(self):
+
+        Returns
+        -------
+
+            processed_data : dataframe
+            Dataframe of dHdl or u_nk data processed using automated equilibration
+            detection followed by statistical inefficiency.
+        """
+
+        # Subsample according to equilibration detection followed by statistical inefficiency.
+        eq_okay = False
+        sample_okay = False
+        try:
+            eq_data = [_equilibrium_detection(i, i.iloc[:, 0])
+                       for i in data]
+            eq_okay = True
+            sampled_data = [_statistical_inefficiency(i, i.iloc[:, 0])
+                            for i in eq_data]
+            sample_okay = True
+        except:
+            pass
+        
+        # Throw errors if either failed
+        if not eq_okay:
+            _warnings.warn("Could not detect equilibration.")
+            try:
+                sampled_data = [_statistical_inefficiency(i, i.iloc[:, 0])
+                                for i in data]
+                sample_okay = True
+            except:
+                _warnings.warn("Could not calculate statistical inefficiency.")
+                sampled_data = data
+
+        if eq_okay and not sample_okay:
+            _warnings.warn("Could not calculate statistical inefficiency.")
+            sampled_data = eq_data
+
+        # make sure there are more than 50 samples for the analysis
+        if eq_okay or sample_okay:
+            for i in sampled_data:
+                if len(i.iloc[:, 0]) < 50:
+                    _warnings.warn(
+                        "Less than 50 samples as a result of preprocessing. No preprocessing will be performed.")
+                    sampled_data = data
+
+        # concatanate in alchemlyb format
+        processed_data = _alchemlyb.concat(sampled_data)
+
+        return processed_data
+
+    @staticmethod
+    def analyse_mbar(files, temperatures, lambdas, engine):
+        """Analyse existing free-energy data using MBAR and the alchemlyb library.
+
+           Parameters
+           ----------
+
+           files : list
+               List of files for all lambda values to analyse. Should be sorted.
+
+           temperatures : list
+               List of temperatures at which the simulation was carried out at for each lambda window.
+               Index of the temperature value should match it's corresponding lambda window index in files.
+
+           lambdas : list
+               Sorted list of lambda values used for the simulation.
+
+           engine : str
+               Engine with which the simulation was run.
+
+           Returns
+           -------
+
+           pmf : [(float, :class:`Energy <BioSimSpace.Types.Energy>`, :class:`Energy <BioSimSpace.Types.Energy>`)]
+               The potential of mean force (PMF). The data is a list of tuples,
+               where each tuple contains the lambda value, the PMF, and the
+               standard error.
+
+           overlap : [ [ float, float, ... ] ]
```

Comment:
Just to check... with the new analysis, is the overlap always returned as a numpy matrix? If so, can the docstrings be updated appropriately, i.e. to `overlap : numpy.matrix`.